packages/jaeger-ui-components/src/selectors/span.ts

```diff
@@ -15,50 +15,52 @@
 import fuzzy from 'fuzzy';
 import { createSelector } from 'reselect';
 
+import { TraceSpan, TraceSpanData, TraceSpanReference } from '../types/trace';
+
 import { getProcessServiceName } from './process';
 
-export const getSpanId = (span) => span.spanID;
-export const getSpanName = (span) => span.operationName;
-export const getSpanDuration = (span) => span.duration;
-export const getSpanTimestamp = (span) => span.startTime;
-export const getSpanProcessId = (span) => span.processID;
-export const getSpanReferences = (span) => span.references || [];
+export const getSpanId = (span: TraceSpanData) => span.spanID;
+export const getSpanName = (span: TraceSpanData) => span.operationName;
+export const getSpanDuration = (span: TraceSpanData) => span.duration;
+export const getSpanTimestamp = (span: TraceSpanData) => span.startTime;
+export const getSpanProcessId = (span: TraceSpanData) => span.processID;
+export const getSpanReferences = (span: TraceSpanData) => span.references || [];
 export const getSpanReferenceByType = createSelector(
-  createSelector(({ span }) => span, getSpanReferences),
-  ({ type }) => type,
-  (references, type) => references.find((ref) => ref.refType === type)
+  createSelector(({ span }: { span: TraceSpanData }) => span, getSpanReferences),
+  ({ type }: { type: string }) => type,
+  (references, type) => references.find((ref: TraceSpanReference) => ref.refType === type)
 );
 export const getSpanParentId = createSelector(
-  (span) => getSpanReferenceByType({ span, type: 'CHILD_OF' }),
+  (span: TraceSpanData) => getSpanReferenceByType({ span, type: 'CHILD_OF' }),
   (childOfRef) => (childOfRef ? childOfRef.spanID : null)
 );
 
-export const getSpanProcess = (span) => {
-  if (!span.process) {
+export const getSpanProcess = (span: TraceSpan | TraceSpanData) => {
+  if ('process' in span) {
```

Comment:
This changes the functionality here. If `span.process = undefined` before it would throw but here `"process" in span === true` so this function would return undefined which probably isn't intended to.

Reply:
I have now added an extra conditional so that `span.process` is only returned if it has a truthy value. https://github.com/grafana/grafana/pull/58006/commits/d0a162f8cb25878990fbdae23fdd30d6c081e0ad.

Reply:
At that point why do we need both `'process' in span && span.process`? `span.process` should be enough shouldn't it?

Reply:
This function takes a span of type `TraceSpan` or `TraceSpanData`. An object of type `TraceSpanData` doesn't need to have a `process` property so simply doing `span.progress` results in the following error:
```vscode
Property 'process' does not exist on type 'TraceSpanData'
```
There, we need to first check if `span` contains a property called `process`.

Reply:
Hmm you are right in that case `'process' in span && span.process` makes sense to assert we are working with `TraceSpan` type. In that case is those ever used with `TraceSpanData` anywhere? Seems like that would inevitably always fail with exception so it seems like it makes more sense to have that as a compile time type constraint.

Reply:
It was used with `TraceSpanData` in at least one place which was a test but I've made some changes so that this function no longer needs to accept a span of type `TraceSpanData`. See commit https://github.com/grafana/grafana/pull/58006/commits/3bd20eb496f5c92ef088ece4d670062df81f794c.